compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -610,6 +616,57 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
         }
     }
 
+    if tcx.features().struct_target_features
+        && let Some(sig) = fn_sig_outer()
+    {
+        // Collect target features from types reachable from arguments.
+        // We define a type as "reachable" if:
+        //  - it is a function argument
+        //  - it is a field of a reachable struct
+        //  - there is a reachable reference to it
+        // FIXME: we may want to cache the result of this computation.
+        let mut visited_types = FxHashSet::default();
+        let mut reachable_types: Vec<_> = sig.skip_binder().inputs().skip_binder().to_owned();
+        let mut additional_tf = vec![];
+
+        while let Some(ty) = reachable_types.pop() {
+            if visited_types.contains(&ty) {
+                continue;
+            }
+            visited_types.insert(ty);
+            if ty.is_ref() {
```

Comment:
I think it would make sense to exhaustively match to make sure you're not missing anything

Reply:
Not sure how to best do that given that `ty.kind()` has ~25 variants - do you propose to explicitly match on all of them, or am I missing something?